config/acme/machines/config_batch.xml

```diff
@@ -454,8 +461,10 @@
     <job name="case.st_archive">
       <template>template.st_archive</template>
       <task_count>1</task_count>
-      <!-- If DOUT_S is true and case.run (or case.test) exits successfully then run st_archive-->
-      <dependency>case.run or case.test</dependency>
+      <!-- If DOUT_S is true and case.run amd/or case.test exits (regardless of success) then run st_archive
+	   This dependence is 'and' if both are run before case.st, and is 'or' if one is not run
+      -->
+      <dependency>case.run case.test</dependency>
```

Comment:
This was the logic:
`for dep in deps:
  if dep in depid.keys() and depid[dep] is not None:`
depid is a map containing the job names which were started previously. There wasn't anything handling the logical statements; just some commented out code below saying that we doubted they would be used.